rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -689,1112 +552,1189 @@ fn induct_loopback_stream_with_memory_limit_impl(
+            // The expected outgoing stream is empty with `begin` avanced.
+            let expected_stream = stream_from_config(StreamConfig {
+                begin: 22,
+                ..StreamConfig::default()
+            });
+            // The stream to the migration subnet is new and has 1 reject response in it
+            // due to the other reject signal.
+            let migration_stream = stream_from_config(StreamConfig {
+                messages: vec![generate_reject_response_for(
+                    RejectReason::OutOfMemory,
+                    request_in_stream(state.get_stream(&REMOTE_SUBNET), 21),
+                )],
+                ..StreamConfig::default()
+            });
+            expected_state.with_streams(btreemap![
+                REMOTE_SUBNET => expected_stream,
+                CANISTER_MIGRATION_SUBNET => migration_stream,
+            ]);
+
+            // Act and compare to expected.
+            let mut available_guaranteed_response_memory =
+                stream_handler.available_guaranteed_response_memory(&state);
+            let inducted_state = stream_handler.garbage_collect_local_state(
+                state,
+                &mut available_guaranteed_response_memory,
+                &slices,
+            );
+            assert_eq!(expected_state, inducted_state);
+            assert_eq!(
+                available_guaranteed_response_memory,
+                stream_handler.available_guaranteed_response_memory(&inducted_state),
+            );
+
+            // 1 reject response failed to induct.
+            metrics.assert_inducted_xnet_messages_eq(&[(
+                LABEL_VALUE_TYPE_RESPONSE,
+                LABEL_VALUE_CANISTER_MIGRATED,
+                1,
+            )]);
+
+            // No critical errors raised.
+            metrics.assert_eq_critical_errors(CriticalErrorCounts::default());
+        },
+    );
 }
 
 /// Calls `induct_stream_slices()` with one stream slice coming from `CANISTER_MIGRATION_SUBNET`
-/// as input containing 3 messages, a reject response for a request sent to `REMOTE_CANISTER` from
-/// `LOCAL_CANISTER`, a normal response and a normal request with the same recipients.
+/// as input containing 3 messages:
+/// - a reject response for a request sent to `REMOTE_CANISTER` from `LOCAL_CANISTER`,
+/// - a normal response with the same recipients.
+/// - and a request with the same recipients.
 ///
 /// `LOCAL_CANISTER` is marked as having migrated from `CANISTER_MIGRATION_SUBNET` to
 /// `LOCAL_SUBNET`.
 ///
-/// Tests that this results in the reject response accepted and inducted due the exception in
-/// `StreamHandler::should_accept_message_from()` and the normal response and request accepted
-/// and dropped.
+/// Tests that this results in the reject response accepted and inducted due to the exception in
+/// `StreamHandler::should_accept_message_from()` and the normal response and request are accepted
+/// but dropped.
```

Comment:
I must have written this test initially, but I can't for the life of me understand why we're dropping messages @1 and @2. Could you please clarify that in the doc comment?

Reply:
Changed the doc comment as discussed offline.